pkg/agents/team3/foodCalculations.go

```diff
@@ -2,17 +2,83 @@ package team3
 
 import (
 	"math"
-
-	"github.com/SOMAS2021/SOMAS2021/pkg/utils/globalTypes/food"
 )
 
-/*
-stubbornness range = 0-100
-mood range = 0-100
-morality range = 0-100
-*/
+//To Do:
+//1. Remove int() from platFood once platFood is type int
+
+func foodReqCalc(a *CustomAgent3, targetHP int) int {
+	decayCurrentHP := int((float64(targetHP+a.HealthInfo().HPLossBase)-float64(a.HealthInfo().WeakLevel)*a.HealthInfo().HPLossSlope)/(1.0-a.HealthInfo().HPLossSlope) + 1)
+	return int(-a.HealthInfo().Tau * math.Log(1.0-(float64(decayCurrentHP-a.HP())/a.HealthInfo().Width)))
+}
+
+func targetHPCalc(a *CustomAgent3) int { //10hp -> target is 20hp, 65hp -> target is 65hp, 100hp -> target is 93hp
+	return int(12.0 + 9.0*float64(a.HP())/11.0)
```

Comment:
What are these magic numbers?

Reply:
They are for us to decide what our target HP is (for now). It's trying to avoid a selfish agent from getting all the food since we don't cap food taken

Reply:
They might change for the last agent